### PR TITLE
[C247C-1765] Added code for fetching Memory total width, data width in the library

### PR DESCRIPTION
**Pull request by pritamnegi-continuum**

### CODE REVIEW

File: smbios/decoder.go [hunk 1]
```diff
@@ -265,6 +265,15 @@ func (d *Decoder) next() (*Structure, error) {
 			}
 		}
 
+		if memInfo.TotalWidth > 0 {
+			totalWidth := uint64(memInfo.TotalWidth)
```
Comment (KedarB) [inline smbios/decoder.go]: You do not need a totalWidth.

File: smbios/decoder.go [hunk 2]
```diff
@@ -265,6 +265,15 @@ func (d *Decoder) next() (*Structure, error) {
 			}
 		}
 
+		if memInfo.TotalWidth > 0 {
+			totalWidth := uint64(memInfo.TotalWidth)
+			physicalMemory.TotalWidth = totalWidth
+		}
+		if memInfo.DataWidth > 0 {
+			dataWidth := uint64(memInfo.DataWidth)
```
Comment (KedarB) [inline smbios/decoder.go]: Do not use dataWidth

File: smbios/decoder.go (hunk 1 above)
Comment (pritamnegi-continuum) [reply smbios/decoder.go]: Memory Total Width value has to be extracted. Hence added a check for the same.

Comment (pritamnegi-continuum) [reply smbios/decoder.go]: Ok will remove the temporary variables

File: smbios/decoder.go (hunk 2 above)
Comment (pritamnegi-continuum) [reply smbios/decoder.go]: Ok will remove the temporary variable

**Review by KedarB — COMMENTED** [review]
Please check comments

File: smbios/decoder.go (hunk 1 above)
Comment (pritamnegi-continuum) [reply smbios/decoder.go]: Removed the temporary variables

File: smbios/decoder.go (hunk 2 above)
Comment (pritamnegi-continuum) [reply smbios/decoder.go]: Removed the temporary variables

**Review by KedarB — APPROVED** [review]

**Review by nileshchate — APPROVED** [review]

**Pull request merged**